### DISCUSSION TITLE
Hidden fields

### DISCUSSION THREAD

**Discussion by mbierma** [plangrid/flask-rebar#296] [Ideas]
We have a few use cases where hidden fields are useful -- things like:

- testing out feature flags that aren't intended for a wider audience
- excluding deprecated/specific use fields from certain specifications

 Hidden endpoints were released in https://github.com/plangrid/flask-rebar/pull/191, but there isn't currently support for hidden fields. Is there any appetite for hidden fields for the wider audience? Something similar to:
 
 ```python
class FooSchema(m.Schema):

    secret = m.fields.String(hidden=True)
```

**Comment by gtmanfred** [comment]
There would require some overhaul for that to happen, because flask rebar does not provide the fields objects. those all come from marshmallow. You can implement this yourself by creating the fields that you need and registering them to the convertors and just returning `{}` as part of the swagger conversion. But I don't think there is a desire to add that widely to flask-rebar.

**Comment by mbierma** [comment]
Right now, we aren't creating the fields themselves, we're just using the field `metadata`, with something like:

```python
super_secret = fields.String(metadata={"hidden": True})
```

and

```python
def get_schema_fields(schema):
    """Retrieve all the names and field objects for a marshmallow Schema
    """
    fields = []
    for name, field in schema.fields.items():
        if 'hidden' in field.metadata and field.metadata['hidden']:
            continue
        prop = compat.get_data_key(field)
        fields.append((prop, field))
    return sorted(fields)
```

**Reply by gtmanfred** [reply]
Yeah, I would be down for something like that, as long as we can do that without having to supply our own String fields for Marshmallow.  Would you be willing to open a PR to add something like that so I can bring it up with the wider team?

Thanks,
Daniel
